Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,107 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Background;
+
+namespace Microsoft.Toolkit.Uwp.Helpers
+{
+    /// <summary>
+    /// This class provides static helper methods for background task.
+    /// </summary>
+    public static class BackgroundTaskHelper
+    {
+        /// <summary>
+        /// Check if a background task has been registered with the application or not.
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task.</param>
+        /// <returns>True/False indicating if a background task was registered or not</returns>
+        public static bool IsBackgroundTaskRegistered(Type backgroundTaskType)
+        {
+            return BackgroundTaskRegistration.AllTasks.Any(t => t.Value.Name == backgroundTaskType.Name);
+        }
+
+        /// <summary>
+        /// Register a background task for the application with conditions
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task</param>
+        /// <param name="trigger">Trigger that indicate when the background task should be invoked</param>
+        /// <param name="enforceConditions">Indicate if the background task should quit if condition is no longer valid</param>
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static async Task<BackgroundTaskRegistration> RegisterAsync(Type backgroundTaskType, IBackgroundTrigger trigger, bool enforceConditions = true, params IBackgroundCondition[] conditions)
+        {
+            // Check if the task is already registered.
+            if (IsBackgroundTaskRegistered(backgroundTaskType) == true)
+            {
+                return GetBackgroundTask(backgroundTaskType) as BackgroundTaskRegistration;
+            }
+
+            // Verify access
+            BackgroundAccessStatus taskRequired = await BackgroundExecutionManager.RequestAccessAsync();
```

Comment:
@ScottIsAFool  @deltakosh Should we even be checking this? Or should we just blindly continue?

At first I thought, yes, check it. However, after more thought... I think we shouldn't. And here's why.

Say the user Bob installs my app, and then they go into the system settings and disable background access for my app. Then they launch my app. I attempt to register a periodic background task that runs every 30 minutes. It's not a critical task - my app functions fine without it - it's just something beneficial.

What happens when Bob decides to re-enable background access for my app? He actually has to launch my app, so that my app can register the background task, since currently my app would have failed to register since Bob previously denied access when my app ran.

Point is - always allow the background task to be registered. If the background task is a critical component of the app, the app developer should check the BackgroundAccessStatus themselves, and then notify the user "Hey, background tasks are important, my app won't be functional without them".

And always registering means that the second the user re-enables background tasks for the app, the background tasks will be live.